bundle/edu.gemini.p2checker/src/main/scala/edu/gemini/p2checker/rules/ghost/GhostRule.scala

```diff
@@ -0,0 +1,46 @@
+package edu.gemini.p2checker.rules.ghost
+
+import edu.gemini.p2checker.api.{IP2Problems, IRule, ObservationElements, P2Problems}
+import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.spModel.core.Coordinates
+import edu.gemini.spModel.gemini.ghost.GhostScienceAreaGeometry
+
+object GhostRule extends IRule {
+  object CoordinatesOutOfFOVRule extends IRule {
+    override def check(elems: ObservationElements): IP2Problems = {
+      val problems = new P2Problems()
+
+      // COORDINATES, i.e. sky positions
+      for {
+        toc <- elems.getTargetObsComponentNode.asScalaOpt
+        ctx <- elems.getObsContext.asScalaOpt
+        env <- Option(ctx.getTargets)
+        base <- env.getAsterism.basePosition(None)
```

Comment:
The time of reference is defined in the `ObsContext` as the scheduling block start.  You can directly get the base position coordinates at the correct time via `ctx.getBaseCoordinates` or you can fish out the `getSchedulingBlockStart`.  In reality for sidereal targets it won't matter because we aren't taking into account proper motion.  Non-sidereal GHOST observing is ??? but nevertheless we should probably use the scheduling block time as intended.